### PR TITLE
Implement netlify-preview action

### PR DESCRIPTION
**Pull request by francoislehoux-okta**
## Description:
- Implement netlify-preview action

To toggle the job, comment `/netlify preview`

This only runs when
- Within okta/okta-developer-docs
- If secrets are not available, the job will short-circuit
- Triggers on new PR comments
- Executes if PR comment is "/netlify preview"

### Resolves:
* [OKTA-558239](https://oktainc.atlassian.net/browse/OKTA-558239)

Screen Shot 2022-12-15 at 7 04 38 PM

### CODE REVIEW

File: .github/workflows/netlify-preview.js.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      uses: nwtgck/actions-netlify@v2.0.0
+      with:
+        publish-dir: 'packages/@okta/vuepress-site/dist'
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        deploy-message: "Deploy from GitHub Actions"
+        enable-pull-request-comment: true
+        overwrites-pull-request-comment: true
+      env:
+        NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+        NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
```
Comment (francoislehoux-okta) [inline .github/workflows/netlify-preview.js.yml:42]: I thought I could make this faster than it used to by removing but uploading `>2500` files to Netlify via `netlify-cli` (which is what this action uses) takes between 10 to 15 minutes.

File: .github/workflows/netlify-preview.js.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: 'Netlify'
+
+on:
+  pull_request:
+    types: [labeled]
+
+jobs:
+  deploy-preview:
+    if: github.repository_owner == 'okta' && github.event.label.name == 'preview'
```
Comment (francoislehoux-okta) [inline .github/workflows/netlify-preview.js.yml:9]: Makes it run only if on `okta/okta-developer-docs`, forks won't work.

File: .github/workflows/netlify-preview.js.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: 'Netlify'
+
+on:
+  pull_request:
+    types: [labeled]
```
Comment (francoislehoux-okta) [inline .github/workflows/netlify-preview.js.yml:5]: This event will trigger the action only when we add a label to the PR. Below we execute the job only if that new label matches `preview`.

File: .github/workflows/netlify-preview.js.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    - name: Build preview
+      run: yarn preview
+    - run: ls
+    - name: Deploy preview to Netlify
+      uses: nwtgck/actions-netlify@v2.0.0
+      with:
+        publish-dir: 'packages/@okta/vuepress-site/dist'
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        deploy-message: "Deploy from GitHub Actions"
+        enable-pull-request-comment: true
```
Comment (francoislehoux-okta) [inline .github/workflows/netlify-preview.js.yml:38]: This writes the deployment link through a comment

File: .github/workflows/netlify-preview.js.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        overwrites-pull-request-comment: false
```
Comment (francoislehoux-okta) [inline .github/workflows/netlify-preview.js.yml:39]: Setting this one to false so it writes a new comment for every deployment

File: package.json [hunk 6]
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "dev": "yarn workspace @okta/vuepress-site dev",
     "build": "yarn stylelint && yarn workspace @okta/vuepress-site build",
+    "preview": "yarn workspace @okta/vuepress-site build",
```
Comment (shawyu-okta) [inline package.json:9]: maybe `build` can use the `preview` script?

**Review by shawyu-okta — APPROVED** [review]

File: package.json (hunk 6 above)
Comment (francoislehoux-okta) [reply package.json:9]: Wdym?

Tbh this doesn't save much time, I'd be curious to see the difference once eslint is in there too.

Comment (shawyu-okta) [reply package.json:9]: i was thinking
```
    "build": "yarn stylelint && yarn preview",
```

File: .github/workflows/netlify-preview.js.yml (hunk 4 above)
Comment (francoislehoux-okta) [reply .github/workflows/netlify-preview.js.yml:36]: Moved to a manual trigger (button click)

**Comment by francoislehoux-okta** [comment]
Test

**Pull request merged**